haskell/defs.bzl

```diff
@@ -253,27 +228,83 @@ haskell_library = rule(
     ],
     fragments = ["cpp"],
 )
-"""Build a library from Haskell source.
 
-Example:
-  ```bzl
-  haskell_library(
-      name = "hello-lib",
-      srcs = glob(["src/**/*.hs"]),
-      src_strip_prefix = "src",
-      deps = [
-          "//hello-sublib:lib",
-      ],
-      reexported_modules = {
-          "//hello-sublib:lib": "Lib1 as HelloLib1, Lib2",
-      },
-  )
-  ```
+def _haskell_worker_wrapper(rule_type, **kwargs):
+    defaults = dict(
+        worker = select({
+            "@rules_haskell//haskell:use_worker": Label("@rules_haskell//tools/worker:bin"),
+            "//conditions:default": None,
+        }),
+    )
+    defaults.update(kwargs)
+
+    if rule_type == 1:
```

Comment:
Better use strings as tags, say `"binary"`, `"test"`, and `"library"` rather than numbers. `rule_type = "binary"` is much more descriptive than `rule_type = 1`.